### PR TITLE
Closed #146 - Write tests

### LINKED ISSUE #161: Update Unit tests
- autofixture
- fluent assertions

### LINKED ISSUE #190: Clean up project
Delete Car folder

### PR DESCRIPTION
**Pull request by strelovidniy**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [x] @github_username

### Second Level Review

- [x] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

File: Car.Backend/Car.UnitTests/Controllers/UserControllerTest.cs [hunk 1]
```diff
@@ -15,18 +15,18 @@ namespace Car.UnitTests.Controllers
...
         {
             userService = new Mock<IUserService>();
-            imageService = new Mock<IImageService<Data.Entities.User, File>>();
+            imageService = new Mock<IImageService<User, File>>();
             userController = new UserController(userService.Object, imageService.Object);
         }
 
-        public Data.Entities.User GetTestUser() =>
-            new Data.Entities.User()
+        public User GetTestUser() =>
```
Comment (OleksandrAntoniuk) [inline Car.Backend/Car.UnitTests/Controllers/UserControllerTest.cs]: please remove this method and use Fixture to create a model

File: Car.Backend/Car.UnitTests/Controllers/UserControllerTest.cs [hunk 2]
```diff
@@ -84,9 +84,9 @@ public void TestGetUserWithAvatarById_WithRightId_ReturnsOkObjectResult()
             using (new AssertionScope())
```
Comment (OleksandrAntoniuk) [inline Car.Backend/Car.UnitTests/Controllers/UserControllerTest.cs:84]: we don't need multiple assertions
check these lines
_((result as OkObjectResult)?.Value as User)?.Id.Should().Be(user.Id);
((result as OkObjectResult)?.Value as User)?.Name.Should().Be(user.Name);_
would be enough

File: Car.Backend/Car.UnitTests/Controllers/UserControllerTest.cs [hunk 3]
```diff
@@ -121,7 +121,7 @@ public async Task UploadUserAvatar_WhenUserNotExist_ReturnsNull()
             // Act
             imageService
                 .Setup(p => p.UploadImage(user.Id, formImage.Image))
-                .Returns(Task.FromResult<Data.Entities.User>(null));
+                .Returns(Task.FromResult<User>(null));
 
             // Assert
             using (new AssertionScope())
```
Comment (OleksandrAntoniuk) [inline Car.Backend/Car.UnitTests/Controllers/UserControllerTest.cs:127]: omit this usage

**Review by OleksandrAntoniuk — CHANGES_REQUESTED** [review]

**Review by IhorSolovii — APPROVED** [review]

**Review by anastasiiakolomyi — APPROVED** [review]
good

**Pull request merged**